bioblend/galaxy/objects/wrappers.py

```diff
@@ -183,6 +183,7 @@ class Workflow(Wrapper):
     def __init__(self, wf_dict, gi=None):
         super(Workflow, self).__init__(wf_dict, gi=gi)
         missing_ids = []
+        tools_list_by_id = [t.id for t in gi.tools.list()]
```

Comment:
Ah, of course, `self.` is not needed because this is the `__init__()` method and `gi` is also a parameter!